internals/detect_intent.py

```diff
@@ -163,17 +246,43 @@ def set_intent_thread_url(self, feature, approval_field, thread_url):
       feature.intent_to_ship_url = thread_url
       feature.put()
 
-  def create_approvals(self, feature_id, approval_field, from_addr, body):
+  def create_approvals(self, feature, approval_field, from_addr, body):
     """Store either a REVIEW_REQUESTED or an APPROVED approval value."""
-    # Case 1: This is a new intent thread
-    existing_approvals = models.Approval.get_approvals(
-        feature_id=feature_id, field_id=approval_field.field_id)
-    if not existing_approvals:
+    feature_id = feature.key.integer_id()
+
+    # Case 1: Detect LGTMs in body, verify that sender has permission,
+    # set an approval value, and clear the original REVIEW_REQUESTED if
+    # the approval rule (1 or 3 LTGMs) is satisfied.
+    if (detect_lgtm(body) and
+        is_lgtm_allowed(from_addr, feature, approval_field)):
+      logging.info('found LGTM')
+      models.Approval.set_approval(
+          feature_id, approval_field.field_id,
+          models.Approval.APPROVED, from_addr)
+      self.record_lgtm(feature, approval_field, from_addr)
+
+    # Case 2: Create a review request for any discussion that does not already
+    # have any approval values stored.
+    elif detect_new_thread(feature_id, approval_field):
+      logging.info('found new thread')
       models.Approval.set_approval(
           feature_id, approval_field.field_id,
           models.Approval.REVIEW_REQUESTED, from_addr)
 
-    # Case 2: This is an existing intent thread
-    # TODO(jrobbins): Detect LGTMs in body, verify that sender has permission,
-    # set an approval value, and clear the original REVIEW_REQUESTED if
-    # the approval rule (1 or 3 LTGMs) is satisfied.
+  def record_lgtm(self, feature, approval_field, from_addr):
+    """Add from_addr to the old way or recording LGTMs."""
+
+    # Note: Intent to prototype has no old field
+    # TODO(jrobbins): Ready-for-trial threads
+
```

Comment:
```suggestion
```

Reply:
Now that the comment is rewritten, I think it should be more clear now that the blank line here should stay.